### PR TITLE
feat: recommendations

### PR DESCRIPTION
**Pull request by gjj**
In this MR, we:
1. created a render template for the Recommendations section of an intern's Certificate of Appreciation
1. using https://github.com/Open-Attestation/decentralized-renderer-react-components/pull/35's new `predicate` feature to selectively display tabs (e.g. if the cert has recommendations, then the `Recommendations` tab will show up)

For point 2, refer to the following demo (both are using this MR's deploy preview):
- [Cert with recommendations](https://dev.opencerts.io/?q=%7B%22type%22:%22DOCUMENT%22,%22payload%22:%7B%22uri%22:%22https://gist.githubusercontent.com/gjj/424c690292c3dde10da3472c3fe92727/raw/cb51ddb059085dce4b29f2123d12f435f2a6fb3f/Document-1-Steven-Koh-ropsten.opencert%22,%22permittedActions%22:%5B%22STORE%22%5D,%22redirect%22:%22https://opencerts.io%22%7D%7D)
- [Cert without recommendations](https://dev.opencerts.io/?q=%7B%22type%22:%22DOCUMENT%22,%22payload%22:%7B%22uri%22:%22https://gist.githubusercontent.com/gjj/424c690292c3dde10da3472c3fe92727/raw/cb51ddb059085dce4b29f2123d12f435f2a6fb3f/Document-2-Barry-Lim-ropsten.opencert%22,%22permittedActions%22:%5B%22STORE%22%5D,%22redirect%22:%22https://opencerts.io%22%7D%7D)

The Recommendations section follows the style of LinkedIn's recommendations.

### CODE REVIEW

**Comment by Nebulis** [comment]
It's ready?

**Comment by gjj** [comment]
@Nebulis in a way yeah:
- schema is done
- however, CSS for the Recommendations is pretty raw

You can download this sample OpenCert with recommendations to try out at dev.opencerts.io: [Document-1-Steven-Koh-ropsten.log](https://github.com/Open-Attestation/govtech-decentralized-renderer/files/5335718/Document-1-Steven-Koh-ropsten.log) (note that I renamed it to `.log` so that I can upload here directly, so ignore the `.log` extension ~)

File: static/index.html [hunk 1]
```diff
@@ -5,41 +5,42 @@
...
-          /////////,          (#####    *////////*               /(((((         
-            .///////////      (((### //////////                  /(((((         
-                 *//////////*.(((((( /////*                         (((         
-                     *///////,(((((( /,                                         
-                         ,///,(((((((,                                          
-                               ((((((((((,                  ,                   
-                                  /((((((((((*         .########*               
-                                      ,((((((((((((((#########/                 
-                                          .(((((((((((((##*                     
+    console.log(`
```
Comment (Nebulis) [inline static/index.html:14]: this is OC right, not OA

Comment (gjj) [reply static/index.html:14]: yeah this is OC!

Comment (Nebulis) [reply static/index.html:14]: it's an OA renderer :eyes:

Comment (gjj) [reply static/index.html:14]: I'll edit this in another MR but yes thanks for pointing that out 😂

File: src/templates/certificateOfAchievement/index.tsx [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 export const templates = [
   {
     id: "CERTIFICATE_OF_ACHIEVEMENT",
     label: "Certificate",
     template: CertificateOfAchievement
+  },
+  {
+    id: "RECOMMENDATIONS",
+    label: "Recommendations",
+    template: Recommendations
```
Comment (Nebulis) [inline src/templates/certificateOfAchievement/index.tsx:13]: did you try with https://github.com/Open-Attestation/decentralized-renderer-react-components/pull/35#pullrequestreview-503528789 ?

File: src/templates/certificateOfAchievement/recommendations.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Meta, Preview, Props, Description, Story } from "@storybook/addon-docs/blocks";
+import { object } from "@storybook/addon-knobs";
+import { Recommendations } from "./recommendations";
+import { sampleCertificate } from "../sample";
+
+<Meta title="MDX|CustomTemplate" component={Recommendations} />
```
Comment (Nebulis) [inline src/templates/certificateOfAchievement/recommendations.stories.mdx:6]: Change title :)

File: src/templates/certificateOfAchievement/recommendations.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import React, { FunctionComponent } from "react";
+import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";
+import { css } from "@emotion/core";
+import { GovTechCertificateTemplate } from "../sample";
+import { AllHtmlEntities } from "html-entities";
```
Comment (Nebulis) [inline src/templates/certificateOfAchievement/recommendations.tsx:5]: Why do we need this ?

Comment (gjj) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: special characters like `'` are rendered as encoded html entities like `&#39;` by react

Comment (Nebulis) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: We use `entities.decode(item.recommendation)`. According to the doc `console.log(entities.decode('&lt;&gt;&quot;&amp;&copy;&reg;')); // <>"&©®` which makes me assume we have entities in the string and react is not involved at all :thinking:

Comment (gjj) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: actually yeah... I was gonna test sth while doing this, like trying to insert some malicious JS code into an OpenAttestation file, just to see if it'll be executed by the renderer

just an idea now, will try it soon

Comment (Nebulis) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: https://reactjs.org/docs/dom-elements.html

Comment (gjj) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: thanks! will take a look

Comment (gjj) [reply src/templates/certificateOfAchievement/recommendations.tsx:5]: ok so I checked the `html-entities` library, it doesn't have a feature to allow only select characters.

I'll check out the `he` library:
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/he/index.d.ts

File: src/templates/certificateOfAchievement/index.tsx (hunk 2 above)
Comment (gjj) [reply src/templates/certificateOfAchievement/index.tsx]: done! updated the main MR message with demo

File: package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
   "dependencies": {
     "@emotion/core": "^10.0.35",
     "@emotion/styled": "^10.0.27",
-    "@govtechsg/decentralized-renderer-react-components": "^2.3.3",
+    "@govtechsg/decentralized-renderer-react-components": "git+https://github.com/Open-Attestation/decentralized-renderer-react-components.git#feat/renderer-templates",
```
Comment (Nebulis) [inline package.json:10]: :eyes:

File: src/templates/certificateOfAchievement/certificateOfAchievement.tsx [hunk 6]
```diff
@@ -2,36 +2,48 @@ import React, { FunctionComponent } from "react";
 import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";
 import { css } from "@emotion/core";
 import { format } from "date-fns";
-import { GovTechCertificateTemplate } from "../sample";
+import { GovTechCertificateTemplate } from "./sample";
 import certificateBase from "../../core/certificate-base.png";
 
+// once opencerts adhere to bootstrap's container class, max-width will need to change to align
+export const containerCustom = (): string => {
```
Comment (Nebulis) [inline src/templates/certificateOfAchievement/certificateOfAchievement.tsx:9]: move this into a util file ?

Comment (gjj) [reply src/templates/certificateOfAchievement/certificateOfAchievement.tsx]: yes done! in fact I refactored everything else too.

I moved the type definition of the cert to `types.tsx`

felt weird to have it in a "sample" cert

**Review by Nebulis — APPROVED** [review]

**Pull request merged**